micrometer-core/src/main/java/io/micrometer/core/instrument/binder/httpcomponents/PoolingHttpClientConnectionManagerMetricsBinder.java

```diff
@@ -0,0 +1,222 @@
+            .register(registry);
+        Gauge.builder(METER_TOTAL_PENDING,
+            connectionManager,
+            (connectionManager) -> connectionManager.getTotalStats().getPending())
+            .description(METER_TOTAL_PENDING_DESC)
+            .tags(tags)
+            .register(registry);
+        Gauge.builder(METER_DEFAULT_MAX_PER_ROUTE,
+            connectionManager,
+            PoolingHttpClientConnectionManager::getDefaultMaxPerRoute)
+            .description(METER_DEFAULT_MAX_PER_ROUTE_DESC)
+            .tags(tags)
+            .register(registry);
+    }
+
+    private void registerPerRouteMetrics(MeterRegistry registry) {
+        poolRouteMaxGauge = MultiGauge.builder(METER_ROUTE_MAX)
+            .description(METER_ROUTE_MAX_DESC)
+            .tags(tags)
+            .register(registry);
+        poolRouteAvailableGauge = MultiGauge.builder(METER_ROUTE_CONNECTIONS)
+            .description(METER_ROUTE_CONNECTIONS_DESC)
+            .tags(tags).tag(TAG_CONNECTIONS_STATE, "available")
+            .register(registry);
+        poolRouteLeasedGauge = MultiGauge.builder(METER_ROUTE_CONNECTIONS)
+            .description(METER_ROUTE_CONNECTIONS_DESC)
+            .tags(tags).tag(TAG_CONNECTIONS_STATE, "leased")
+            .register(registry);
+        poolRoutePendingGauge = MultiGauge.builder(METER_ROUTE_PENDING)
+            .description(METER_ROUTE_PENDING_DESC)
+            .tags(tags)
+            .register(registry);
+    }
+
+    /**
+     * Updates the list of routes accessed by the monitored http client.
+     *
+     * Call this method periodically if you want to monitor thread pools for
+     * every accessed route (target scheme, host and port). For example:
+     *
+     * <pre>
+     *     &#x40;Bean
+     *     public RouteUpdater routeUpdater(List&#60;PoolingHttpClientConnectionManagerMetricsBinder&#62; metricsBinders) {
+     *         return new RouteUpdater(metricsBinders);
+     *     }
+     *
+     *     private static class RouteUpdater {
+     *         private final HashSet&#60;PoolingHttpClientConnectionManagerMetricsBinder&#62; metricsBinders = new HashSet&#60;&#62;();
+     *
+     *         private RouteUpdater(Collection&#60;PoolingHttpClientConnectionManagerMetricsBinder&#62; metricsBinders) {
+     *             this.metricsBinders.addAll(metricsBinders);
+     *         }
+     *
+     *         &#x40;Scheduled(fixedRate = 10000)
+     *         public void updateRoutes() {
+     *             metricsBinders.forEach(PoolingHttpClientConnectionManagerMetricsBinder::updateRoutes);
+     *         }
+     *     }
+     * </pre>
+     *
+     * BE CAREFUL! Call this if and only if you are sure that your http client
+     * accesses only a limited number of remote schemes, hosts and ports. DO NOT
+     * USE THIS METHOD if your HttpClient accesses a big or unlimited number of
+     * routes (i.e. if the target host depends on the user input).
```

Comment:
Is there any way that we can capture path variables in a template? That would allow us to tag a path with less worry about tag explosion.